core/src/loader.rs

```diff
@@ -579,6 +580,13 @@ impl<'gc> Loader<'gc> {
                     that.set(k, v.into(), &mut activation)?;
                 }
 
+                // Fire the onData method and event.
+                if let Some(display_object) = that.as_display_object() {
+                    if let Some(movie_clip) = display_object.as_movie_clip() {
+                        movie_clip.event_trigger(&mut activation.context, ClipEvent::Data);
+                    }
+                }
```

Comment:
Thinking on this some more, because the `data` event is odd case, maybe we can fire the `onData` directly here, like:
```
if let Some(display_object) = that.as_display_object() {
                    if let Some(movie_clip) = display_object.as_movie_clip() {
                        context.action_queue.queue_actions(
                                movie_clip.into(),
                                ActionType::Method {
                                    that,
                                    "onData",
                                    args: vec![],
                                },
                            false
                            );
                        movie_clip.event_dispatch(&mut activation.context, ClipEvent::Data);
                    }
                }
```
and remove the `ClipEvent::Data => Some("onData"),` in `ClipEvent::method_name`, so that `event_dispatch` doesn't have to contort itself to fire this one case in the correct order.